docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,344 @@
+`Conditions` status field.
+In the opposite scenario where the `NetworkSelectionElement` does not specify the `IPAMClaimReference`
+the IP allocation is not persisted when the pod is removed.
+
+### Testing Details
+
+
+The following scenarios should be covered in testing:
+
+* VM workloads import into OVN-Kubernetes with no changes to the instances network configuration.
+* Imported VM workloads can live-migrate to another node without any additional traffic disruption.
+* 'v1.multus-cni.io/default-network' cannot be changed after the pod was created.
+* It should be possible to configure the pods MAC or the IP address without configuring the other.
+* Detect MAC and IP address conflicts between the requested addresses for a newly created pods and the addresses that
+are already allocated in the network.
+* After configuring custom default gateway and management addresses on a Layer2 UDN the previous default
+IPs can be consumed by workloads(e.g. for 10.0.0.0/16 network create pods with 10.0.0.1 and 10.0.0.2 addresses).
+* Modifying the  default gateway and management addresses on a Layer2 UDN should not be possible after the network
+was created.
+
+The scenarios mentioned above have to cover both IPv4 and IPv6 IP families.
+
+
+### Documentation Details
+
+## Risks, Known Limitations and Mitigations
+
+* Modifying the 'v1.multus-cni.io/default-network' value after the pod was created could have unpredictable
+consequences.
+To mitigate this introduce a Validating Admission Policy described in [Implementation Details](#implementation-details).
+
+* By allowing users to specify the IP and MAC addresses for the pods there is a risk of conflicts.
+To mitigate this OVN-Kubernetes will check that the requested addresses are not currently used in the UDN.
+There is still a risk that the user picks an address that's consumed by something outside of the UDN but that's beyond
+what OVN-Kubernetes controls and can check.
+
+* The dynamic, per-node subnet allocation in Layer3 UDNs, where each node has a unique default gateway and
+management IP, makes user-specified UDN gateway/management IPs and static pod IP/MAC assignments very complex. This
+enhancement will not support Layer3 UDNs.
+
+* BGP support today is limited to cluster UDNs, to ensure a non-NATed traffic for pods with pre-configured addresses
+the user has to use a cluster UDN to configure the network. This is a limitation unrelated to this enhancement
+and it is possible it will be solved in the future.
+
+* By consuming the 'v1.multus-cni.io/default-network' annotation for altering the primary UDNs pod configuration the
+user won't be able to use it for configuring the cluster default network attachment. This is acceptable as there is
+currently no support for modifying the cluster default network through this annotation while using primary UDNs.
+If there is a requirement in the future another mechanism can be considered.
+
```

Comment:
We should add a bullet about default gw and ipv6, we cannot enforce the default gw IPv6 address to VMs since RAs always use the LLA (that is computed from the mac address), this means that VMs will have a ipv6 default gw with two paths:
- original ipv6 gateway before migration (this can be a LLA too, depending on how did they configure)
- the LLA address after migration as result from ovn-k answring the RS with a RA.

Both paths should work since the GW router LRP will have configure also the static ipv6 gateway address, although don't know if this is also a LLA can be a problem.

Reply:
> * riginal ipv6 gateway before migration (this can be a LLA too, depending on how did they configure)

Are you sure the "old" default route learned through RA will persist the VM reboot? I don't think it would.
I've added my reasoning for why we should have the global default GW v6 IP configurable here: 
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5238#discussion_r2097083291